proc/basic_server.cpp

```diff
@@ -13,13 +14,39 @@ void error(const char *msg) {
     exit(EXIT_FAILURE);
 }
 
+void *connection_handler(void *socket_desc) {
+    int sockfd = *(int *)socket_desc;
+    char buf[BUFLEN] = {0};
+    char revbuf[BUFLEN] = {0};
+
+    if(read(sockfd, buf, BUFLEN) < 0) {
+        error("ERROR reading from socket\n");
+    }
+
+    // Reverse string
+    int i;
+    int j = 0;
+    int len = strlen(buf);
+    for(i = len - 1; i >= 0; i--) {
+        revbuf[j++] = buf[i];
+    }
+    revbuf[i] = '\0';
```

Comment:
i should be j because i after for loop is -1. Shouldn't be able to assign nul terminate on -1 position of revbuf.

Reply:
Fixed.